sentry-spring-jakarta/src/test/kotlin/io/sentry/spring/jakarta/tracing/SentryTracingFilterTest.kt

```diff
@@ -117,6 +120,8 @@ class SentryTracingFilterTest {
         verify(fixture.hub).captureTransaction(
             check {
                 assertThat(it.contexts.trace!!.status).isEqualTo(SpanStatus.INTERNAL_ERROR)
+                assertThat(it.extras?.get(SpanDataConvention.HTTP_STATUS_CODE_KEY)).isEqualTo(500)
```

Comment:
`data` of transactions lands in `extras`, is this OK?

Reply:
why does it land in `extras`? shouldn't it be available through `getData()` method?

Reply:
The following code seems to do this on purpose:

https://github.com/getsentry/sentry-java/blob/695d3a3acd91e7759da0699f9b5e004d945de53b/sentry/src/main/java/io/sentry/protocol/SentryTransaction.java#L95-L100

Reply:
oh, i see, i was missing it 😅 
in https://github.com/getsentry/sentry-java/pull/2786 and https://github.com/getsentry/team-sdks/issues/9 we talked about span data, not transaction extra.
AFAIK the goal is to add the data directly to the spans, not to transactions.
Maybe @smeubank can help us removing any doubt?